src/main/scala/sparkz/core/network/message/MessageSerializer.scala

```diff
@@ -40,6 +40,10 @@ class MessageSerializer(specs: Seq[MessageSpec[_]], magicBytes: Array[Byte], mes
     } else {
       val it = byteString.iterator
       val magic = it.getBytes(MagicLength)
+      //peer is from another network
+      if (!java.util.Arrays.equals(magic, magicBytes)) {
+        throw MaliciousBehaviorException(s"Wrong magic bytes, expected ${magicBytes.mkString}, got ${magic.mkString} in : ${byteString.utf8String}")
+      }
       val msgCode = it.getByte
```

Comment:
Let's check more careful the case, when we retrieved the message with the code which node spec registered:
`val spec = specsMap.getOrElse(msgCode, throw new Error(s"No message handler found for $msgCode"))`.

Here I can see that:
1) we may check the spec right now, no need to wait for the full message.
2) will we stuck reading the same message? I don't see the code that removes this data from `chunksBuffer` in `PeerConnectionHandler.scala`. Check if we already have some UTs for this method. If not - let's add  a couple to confirm that it works/doesn't work